plugins/guests/darwin/cap/flavor.rb

```diff
@@ -0,0 +1,24 @@
+module VagrantPlugins
+  module GuestDarwin
+    module Cap
+      class Flavor
+        def self.flavor(machine)
+          # Read the version file
+          output = ""
+          machine.communicate.sudo("sw_vers -productVersion", error_check: false) do |_, data|
```

Comment:
I'm pretty sure `sw_vers` is only a macos thing, but I couldn't find a more generic darwin command to get the flavor for the darwin guest. In this case, I figure it can be lax on error checking and an ok default return value could be `:darwin`